tests/main/interfaces-desktop-launch/task.yaml

```diff
@@ -51,3 +51,13 @@ execute: |
 
     echo "The app was invoked with the arguments in the desktop file"
     MATCH "^args=arg-before arg-after$" < "$launch_data"
+
+    if ! os.query is-core; then
+        exit 0
+    fi
+    echo "The launcher snap can also invoke the snap via the desktop file Exec line"
+    rm "$launch_data"
+    tests.session -u test exec test-launcher.exec \
+        test-app_test-app.desktop
+    retry -n 5 --wait 1 test -s "$launch_data"
+    MATCH "^args=arg-before arg-after$" < "$launch_data"
```

Comment:
we also need to test the negative case of what happens if we run a snap application that does not have desktop file

Reply:
I'm not sure what that test would look like though: if the snap doesn't install a desktop file, then there is nothing for the launcher to parse and try to execute.

Reply:
as I said it'a negative test case, so I expect it to fail and the test should check that it fails and how